src/dc_location.rs

```diff
@@ -49,7 +49,7 @@ impl dc_location {
 #[derive(Clone)]
 pub struct dc_kml_t {
     pub addr: *mut libc::c_char,
-    pub locations: *mut dc_array_t,
+    pub locations: Option<Vec<dc_location>>,
```

Comment:
Why make this optional? should be just an empty vec by default I would think

Reply:
This way `is_null` can be translated directly to `is_none`.

I want to avoid changing any logic for now. Null pointer is translated to `None`, empty `dc_array_t` is translated to empty vec etc.